contracts/src/v0.8/dev/automation/2_1/KeeperRegistryBase2_1.sol

```diff
@@ -172,6 +172,13 @@ abstract contract KeeperRegistryBase2_1 is ConfirmedOwner, ExecutionPrevention {
     OPTIMISM
   }
 
+  enum Trigger {
+    CONDITION,
+    LOG,
+    CRON,
+    READY
```

Comment:
Can we add these when we fully design these? I am not sure what value they add until we think of the e2e design (which might have been done?)

Reply:
e.g. I am not sure how READY is different from condition

Reply:
will need to trigger on type inside of checkUpkeep ex...

```
function checkUpkeep() {
  // do initial balance checks, paused checks, etc...
  // ...
  if type == conditional || type == log {
    return target.checkUpkeep(...)
  } if type == cron {
     return block.timestamp > lastTick // maybe we need? this idk
  } if type == ready {
    return true // no checks for this type
  }
}
```

... also inside of performUpkeep we need to take different actions